fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -0,0 +1,190 @@
+        for value in profile_plot.values():
+            sorted_list = np.sort(value)
+            step_values.append(np.insert(sorted_list, 0, 0.0))
+
+        max_value = np.max([np.max(v)
+                            for v in profile_plot.values()])
+        linear_upper_limit = 10
+
+        use_log_plot = True
+        if max_value < linear_upper_limit:
+            # if we don't need a log plot, then don't print one
+            fig, ax = plt.subplots(1, 2,
+                                   gridspec_kw={
+                                       'width_ratios': [30, 7],
+                                       'wspace': 0.01})
+            legend_ax = 1
+            use_log_plot = False
+        else:
+            fig, ax = plt.subplots(1, 3,
+                                   sharey=True,
+                                   gridspec_kw={
+                                       'width_ratios': [10, 20, 7],
+                                       'wspace': 0.01})
+            legend_ax = 2
+
+        # Plot linear performance profile
+        create_plot(ax[0], step_values, acc.keys())
+        ax[0].set_xlim(1, linear_upper_limit)
+        ax[0].set_xticks([1, 2, 4, 6, 8, 10])
+        ax[0].set_xticklabels(['$1$', '$2$', '$4$', '$6$', '$8$', '$10$'])
+        ax[0].yaxis.set_ticks_position('left')
+
+        if use_log_plot:
+            # Plot log performance profile
+            create_plot(ax[1], step_values, acc.keys())
+            ax[1].set_xlim(
+                linear_upper_limit,
+                min(max_value+1, 10000))
+            ax[1].set_xscale('log')
+            ax[1].set_xticks([100, 1000, 10000])
+            ax[1].set_xticklabels(['$10^2$', '$10^3$', '$10^4$'])
+            ax[1].yaxis.set_ticks_position('right')
+            ax[1].tick_params(axis='y', labelcolor='white')
+
+        # legend
+        ax[legend_ax].axis('off')
+        handles, labels = ax[0].get_legend_handles_labels()
+        wrapped_labels = ['\n'.join(wrap(l, 22)) for l in labels]
+        ax[legend_ax].legend(handles, wrapped_labels, loc=2, prop={'size': 8})
+
+        # Common parts
+        fig.suptitle("{}".format(name))
+        plt.ylim(0.0, 1.0)
+        fig.suptitle("Performance profile - {}".format(name))
+
+        # add a big axis, hide frame
+        fig.add_subplot(111, frameon=False)
+        # hide tick and tick label of the big axis
+        plt.tick_params(labelcolor='none',
+                        top=False,
+                        bottom=False,
+                        left=False,
+                        right=False)
+        plt.xlabel("f                       ")
+        plt.ylabel("fraction for which solver within f of best")
+
+        plt.savefig(this_filename)
+
+    return figure_path
+
+def create_plot(ax, step_values, keys):
+    """
+    Function to draw the profile on a matplotlib axis
+
+    :param ax : A matplotlib axis to be filled
+    :type ax :  an `.axes.SubplotBase` subclass of `~.axes.Axes` \ 
```

Comment:
Trailing whitespace.
Is the backslash necessary for sphinx?

Reply:
fixed